packages/medusa/src/api/routes/admin/batch/create-batch-job.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  IsBoolean,
+  IsJSON,
+  IsOptional,
+  IsString,
+} from "class-validator"
+import BatchJobService from "../../../../services/batch-job"
+import { validator } from "../../../../utils/validator"
+
+/**
+ * @oas [post] /batch
+ * operationId: "PostBatches"
+ * summary: "Create a Batch Job"
+ * description: "Creates a Batch Job."
+ * x-authenticated: true
+ * parameters:
+ *   - (body) type=* {string} The type of batch job to start.
+ *   - (body) context=* {string} Additional infomration regarding the batch to be used for processing.
+ *   - (body) dry_run=* {boolean} Set a batch job in dry_run mode to get some information on what will be done without applying any modifications.
+ * tags:
+ *   - Customer
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *            batch_job:
+ *              $ref: "#/components/schemas/batch_job"
+ */
+export default async (req, res) => {
+  const validated = await validator(AdminPostBatchesReq, req.body)
+
+  const userId = req.user.id ?? req.user.userId
+
+  const batchJobService: BatchJobService = req.scope.resolve("batchJobService")
+  const batch_job = await batchJobService.create({
+    ...validated,
+    created_by: userId,
+  })
+
+  res.status(200).json({ batch_job })
```

Comment:
nit: should we respond with 202

Reply:
good catch

Reply:
In fact are you sure @fPolic? https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202


Since the job is created, i think it would be 201 instead no?

Reply:
yep *201 since a resource has been created and 202 will be when an actual file has been uploaded

Reply:
Yep sounds good 🤘